src/python/pants/backend/makeself/target_types.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2024 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from pants.core.goals.package import OutputPathField
+from pants.engine.target import (
+    COMMON_TARGET_FIELDS,
+    SpecialCasedDependencies,
+    StringField,
+    StringSequenceField,
+    Target,
+)
+from pants.util.docutil import bin_name
+from pants.util.strutil import help_text
+
+
+class MakeselfArthiveLabelField(StringField):
+    alias = "label"
+    help = help_text(
+        """
+        An arbitrary text string describing the package. It will be displayed while extracting
+        the files.
+        """
+    )
+
+
+class MakeselfArchiveStartupScriptField(StringSequenceField):
+    alias = "startup_script"
+    required = False
+    help = help_text(
+        """
+        The startup script, i.e. what gets run when executing `./my_archive.run`.
```

Comment:
Since this is now a sequence of tuples, presumably to allow passing arbitrary arguments, can we add a note or example about that to this docstring?

Reply:
Sure https://github.com/pantsbuild/pants/pull/20360/commits/3a39ee9d045293993570e787cc6ec3912ed21540